source/behaviors/offClick/offClick.ts

```diff
@@ -29,16 +30,36 @@ export class OffClickDirective implements OnInit, OnDestroy {
 	}
 
 	ngOnInit() {
-		setTimeout(() => {
-			document.addEventListener('click', this.listener);
-		}, 0);
+		if (this.active) {
+			setTimeout(() => {
+				this.addListener();
+			});
+		}
 	}
 
-	ngOnDestroy() {
+	ngOnChanges(changes: SimpleChanges): void {
```

Comment:
You don't have hook into change detection like this. Angular does this already. You can add a setter to the input:
`@Input('active')
	set active(value: boolean) { 
		//handle the listener stuff here
	}`